stable/elasticsearch-exporter/templates/servicemonitor.yaml

```diff
@@ -47,4 +47,5 @@ spec:
     - {{ . }}
 {{- end }}
 {{- end }}
+  sampleLimit: {{ .Values.serviceMonitor.sampleLimit }}
```

Comment:
`sampleLimit` default is 0, wouldn't it limit the scrape to 0 samples?

maybe:

```yaml
{{- with .Values.serviceMonitor.sampleLimit }}
sampleLimit: {{ . }}
{{- end }}
```

and have no default value for `sampleLimit`?

Reply:
The docs say that 0 means unlimited (which is the default):
```
# Per-scrape limit on number of scraped samples that will be accepted.
# If more than this number of samples are present after metric relabelling
# the entire scrape will be treated as failed. 0 means no limit.
[ sample_limit: <int> | default = 0 ]
```
This means that `sampleLimit: 0` is equivalent to the current behavior.

Reply:
I don't mind doing it your way as well. So whatever you prefer.